client/modules/IDE/components/FileNode.jsx

```diff
@@ -10,6 +10,47 @@ import FolderRightIcon from '../../../images/triangle-arrow-right.svg';
 import FolderDownIcon from '../../../images/triangle-arrow-down.svg';
 import FileIcon from '../../../images/file.svg';
 
+function FileName({ name }) {
+  const nameArray = name.split('.');
+  if (nameArray.length > 1) {
+    const extension = `.${nameArray[nameArray.length - 1]}`;
+    const fileName = nameArray.slice(0, -1).join('');
+    const firstLetter = fileName[0];
+    const lastLetter = fileName[fileName.length - 1];
+    const middleText = fileName.slice(1, -1);
+    return (
+      <span className="sidebar__file-item-name-text">
+        <span>{firstLetter}</span>
+        {fileName.length > 2 &&
+          <span className="sidebar__file-item-name--ellipsis">{middleText}</span>
+        }
+        {fileName.length > 1 &&
+          <span>{lastLetter}</span>
+        }
+        <span>{extension}</span>
+      </span>
+    );
```

Comment:
If possible, I think `FileName` should be just this block, and the name logic should go on another function, which returns a string. This will avoid having to duplicate this code below.

Reply:
Makes sense!